client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -46,7 +54,75 @@ centroid varying vec2 varTexCoord;
 varying highp vec3 eyeVec;
 varying float nightRatio;
 
+varying vec3 viewVec;
+
 #ifdef ENABLE_DYNAMIC_SHADOWS
+vec4 perm(vec4 x)
+{
+	return mod(((x * 34.0) + 1.0) * x, 289.0);
+}
+
+#if (defined(MATERIAL_WAVING_LIQUID) && defined(ENABLE_WATER_REFLECTIONS)) || defined(ENABLE_BUMPMAPS)
+float snoise(vec3 p)
```

Comment:
snoise is not used in this shader as far as I can tell... Only gnoise

Reply:
True, i think the preprocessor stuff is also not in the right place…